src/pretalx/submission/permissions.py

```diff
@@ -0,0 +1,68 @@
+import rules
+
+from pretalx.person.permissions import is_orga, is_reviewer
+from pretalx.submission.models import SubmissionStates
+
+
+@rules.predicate
+def is_speaker(user, obj):
+    return user in obj.speakers.all()
+
+
+@rules.predicate
+def can_be_withdrawn(user, obj):
+    return SubmissionStates.WITHDRAWN in SubmissionStates.valid_next_states.get(obj.state, [])
+
+
+@rules.predicate
+def can_be_rejected(user, obj):
+    return SubmissionStates.REJECTED in SubmissionStates.valid_next_states.get(obj.state, [])
+
+
+@rules.predicate
+def can_be_accepted(user, obj):
+    return SubmissionStates.ACCEPTED in SubmissionStates.valid_next_states.get(obj.state, [])
+
+
+@rules.predicate
+def can_be_confirmed(user, obj):
+    return SubmissionStates.CONFIRMED in SubmissionStates.valid_next_states.get(obj.state, [])
+
+
+@rules.predicate
+def can_be_canceled(user, obj):
+    return SubmissionStates.CANCELED in SubmissionStates.valid_next_states.get(obj.state, [])
+
+
+@rules.predicate
+def can_be_removed(user, obj):
+    return SubmissionStates.DELETED in SubmissionStates.valid_next_states.get(obj.state, [])
+
+
+@rules.predicate
+def can_be_unconfirmed(user, obj):
+    return SubmissionStates.ACCEPTED in SubmissionStates.valid_next_states.get(obj.state, []) and obj.state == SubmissionStates.CONFIRMED
+
+
+@rules.predicate
+def can_be_edited(user, obj):
+    return obj.editable
+
+
+@rules.predicate
+def can_review_submission(user, obj):
```

Comment:
the _predicate_ here would be `is_not_involved` or `is_not_speaker`, wouldn't it? Or, maybe even better: use the negated `~is_speaker` in `submission.review_submission` and drop this predicate.

Reply:
Excellent point, this is much more readable.